modules/services/OvertureService.js

```diff
@@ -0,0 +1,144 @@
+import { Tiler} from '@rapid-sdk/math';
+
+import { AbstractSystem } from '../core/AbstractSystem.js';
+
+const PMTILES_ROOT_URL = 'https://overturemaps-tiles-us-west-2-beta.s3.us-west-2.amazonaws.com/';
+const PMTILES_CATALOG_PATH = 'pmtiles_catalog.json';
+
+
+/**
+ * `OvertureService`
+ * This service connects to the 'official' sources of Overture PMTiles by acting as a wrapper around the 
+ * vector tile service
+ *
+ * - Protomaps .pmtiles single-file archive containing MVT
+ *    https://protomaps.com/docs/pmtiles
+ *    https://github.com/protomaps/PMTiles
+ *
+ */
+export class OvertureService extends AbstractSystem {
+
+  /**
+   * @constructor
+   * @param  `context`  Global shared application context
+   */
+  constructor(context) {
+    super(context);
+    this.id = 'overture';
+    this.pmTilesCatalog = {};
+
+    this.latestRelease = '';
+
+    // Sources are identified by their URL template..
+    this._sources = new Map();   // Map(template -> source)
+    this._tiler = new Tiler().tileSize(512).margin(1);
+    this._nextID = 0;
+  }
+
+
+ async _loadS3Catalog() {
+  await fetch(PMTILES_ROOT_URL + PMTILES_CATALOG_PATH)
+    .then(response => response.json())
+    .then(catalog => {
+      this.pmTilesCatalog = catalog;
+    })
+    .catch(error => {
+      console.error('Error fetching or parsing the PMTiles STAC Catalog:', error);
+    });
+}
+
+  /**
+   * initAsync
+   * Called after all core objects have been constructed.
+   * @return {Promise} Promise resolved when this component has completed initialization
+   */
+  initAsync() {
+    const context = this.context;
+    const vtService = context.services.vectortile;
+
+    this._loadS3Catalog();
```

Comment:
How you have it written, the catalog fetch isn't tied to this init promise really.
It should maybe be something like:
```js
return Promise.all([
  vtService.initAsync(),
  this._loadS3Catalog()
]); 
```  

Then this service isn't considered "initted" until both things happen.